tensorflow/python/ops/array_grad.py

```diff
@@ -448,3 +451,78 @@ def _MirrorPadGradGrad(op, grad):
 @ops.RegisterGradient("QuantizeAndDequantize")
 def _QuantizeAndDequantizeGrad(_, grad):
   return grad
+
+
+@ops.RegisterGradient("ExtractImagePatches")
+def _ExtractImagePatchesGrad(op, grad):
+
+  batch_size, rows_in, cols_in, channels = [
+    dim.value for dim in op.inputs[0].get_shape()
+  ]
+  _, rows_out, cols_out, _ = [
+    dim.value for dim in op.outputs[0].get_shape()
+  ]
+  _, ksize_r, ksize_c, _ = op.get_attr('ksizes')
+  _, stride_r, stride_h, _ = op.get_attr('strides')
+  _, rate_r, rate_c, _ = op.get_attr('rates')
+  padding = op.get_attr('padding')
+
+  ksize_r_eff = ksize_r + (ksize_r - 1) * (rate_r - 1)
+  ksize_c_eff = ksize_c + (ksize_c - 1) * (rate_c - 1)
+
+  if padding == 'SAME':
+    rows_out = int(ceil(rows_in / stride_r))
+    cols_out = int(ceil(cols_in / stride_h))
+    pad_rows = ((rows_out - 1) * stride_r + ksize_r_eff - rows_in) // 2
+    pad_cols = ((cols_out - 1) * stride_h + ksize_c_eff - cols_in) // 2
+
+  elif padding == 'VALID':
```

Comment:
Looks like in python 3 PIP setups, there is a problem caused by these lines.
It looks like "padding" can be something other than 'SAME' or 'VALID'
any ideas what might be happening?

http://ci.tensorflow.org/view/Tensorflow%20Jenkins%20Monitored%20builds/job/nightly-matrix-cpu/TF_BUILD_CONTAINER_TYPE=CPU,TF_BUILD_IS_OPT=OPT,TF_BUILD_IS_PIP=PIP,TF_BUILD_PYTHON_VERSION=PYTHON3,label=cpu-slave/194/consoleFull